src/main/java/com/iexec/worker/compute/ComputeController.java

```diff
@@ -29,15 +30,41 @@
 import static org.springframework.http.ResponseEntity.ok;
 
 @RestController
-public class PostComputeController {
+public class ComputeController {
 
+    private final ComputeExitCauseService computeStageExitService;
     private final ResultService resultService;
 
-    public PostComputeController(ResultService resultService) {
+    public ComputeController(ComputeExitCauseService computeStageExitService,
+                             ResultService resultService) {
+        this.computeStageExitService = computeStageExitService;
         this.resultService = resultService;
     }
 
-    @PostMapping(path = "/iexec_out/{chainTaskId}/computed")
+    @PostMapping("/compute/{stage}/{chainTaskId}/exit")
+    public ResponseEntity sendExitCauseForGivenComputeStage(
+            @PathVariable String stage,
+            @PathVariable String chainTaskId,
+            @RequestBody ExitMessage exitMessage) {
+        if (!ComputeStage.isValid(stage) || exitMessage.getCause() == null) {
+            return ResponseEntity
+                    .status(HttpStatus.BAD_REQUEST.value())
+                    .build();
+        }
+        if (computeStageExitService.setExitCause(stage,
```

Comment:
This does not seem to be a case that can happen.
This endpoints seems to only be called once from pre-compute and post-compute.

Even if it happens, do we really want to have a slow memory leak in the worker ?
I think I would prefer to report several times the data to the scheduler and add security there if required to keep the worker as simple as possible.

Reply:
No sure to understand your "leak"?

The data isn't directly sent to the scheduler, it is written locally to the worker.
Do you prefer to accept any future update and to overwrite previous value?

Reply:
If I understand correctly, the sendExitCauseForGivenComputeStage is only called once from pre/post compute.
Thus computeStageExitService.setExitCause should only be called once.
We should never be in the case of `ALREADY_REPORTED`

In `ComputeStageExitService`, we have an ever increasing `HashMap` which will infinitely increase in size, this is the memory leak.

Shouldn't the if condition be inverted ?
The method returns `true` if the cause is added to the map and `false` if it was already reported.

Reply:
I agree, in a classic workflow it should be only called once from pre/post compute, I'm here considering error scenarios (bugs, hack, etc) but "Even if it happens, do we really want to have a slow memory leak in the worker ?" I don't get your point about 

I rather see a memory leak on classic flow, since map is never pruned

Reply:
What about using an ExpiringMap there? :)